### DISCUSSION TITLE
Ethereum-Provider  and Sepolia Error Connection

### DISCUSSION THREAD

**Discussion by EgoInc** [WalletConnect/walletconnect-monorepo#3369] [Developer Support]
Hello, I'm trying to migrate from web3-provider to ethereum-provider

### My code 
for authorization page:

```JS
export const provider = await EthereumProvider.init({
        projectId: 'PROJ-ID',
        chains: [11155511],
        showQrModal: true
 })

.....

await provider.connect();
```
at the page with smart-contract interaction:
```JS
 import {provider} from ...

//Connect to ethers
await provider.enable();
lib = new ethers.providers.Web3Provider(provider);

//Connect to 
var contractERC1155 = new ethers.Contract(contract, configErc1155, lib.getSigner());
var approval = await contractERC1155.isApprovedForAll(loggedUser, TraderAddress)
```

### 1. When I start connection with this provider, I haven't option "metamask" on dekstop variants, however it's installed in my browser: 
![image](https://github.com/WalletConnect/walletconnect-monorepo/assets/56895607/446983f0-18c9-472b-a4b2-8731d72ee0fb)

Is there some strings that I should add in EthereumProvider initialization to add this wallet?

### 2. When I use Sepolia network (11155111) - interaction with smart contract throw such error: 

```
Failed to load resource:  https://rpc.walletconnect.com/v1?chainId=eip155:11155111&projectId={_here is my project id_} the server responded with a status of 400 (Bad Request)
```

However when I changed network to Mumbai (80001) - the code starts working without any problems. So what's the problem with Sepolia network?

Thanks in advance =)

**Comment by glitch-txs** [comment]
At the moment only Web3Modal v2 supports browser wallets
https://docs.walletconnect.com/2.0/web/about-web3modal

If the issue with the RPC persists could you open an issue in the monorepo? Thanks 🙏

**Reply by EgoInc** [reply]
@glitch-txs sorry, I didn't catch about provider, as far as I understand Metamask Extention is also a browser wallet. Also when I use wagmi provider for wallet connect authorization - it shows metamask option to me: 
![image](https://github.com/WalletConnect/walletconnect-monorepo/assets/56895607/c59a2d5a-41e2-4457-b89c-f0a65ec172fc)
(I show screen previously that with Ethereum Provider there is not such option even when Metamask is installed in browser) 

So, am I correctly understand - when I use Ethereum Provider for WC authorization there will not be option "metamask extention" for my users and I should use another provider? If it's true, is there some providers for WC V2 that I can use with ethers.js?

**Reply by glitch-txs** [reply]
hey, taking a closer look it seems like you're missing a number in Sepolia chain id `11155111` I also tested connecting to MetaMask and works.

Answering you question yes, we're currently working on Web3Modal v3 that's going to support ethers.js and also browser/extension wallets.

**Reply by glitch-txs** [reply]
here's my repo for testing ethereum-provider btw, I added Sepolia as required chain, you can test it on the vercel link
https://github.com/glitch-txs/walletconnect-cafe
let me know if works for you!

**Reply by EgoInc** [reply]
> we're currently working on Web3Modal v3 that's going to support ethers.js and also browser/extension wallets

@glitch-txs Is the approximate release date of the third version with ethers and browser wallets support?
